Python/Product/PythonTools/PythonTools/Navigation/NavigateTo/PythonNavigateToItemProvider.cs

```diff
@@ -163,34 +163,48 @@ out projectType
             try {
                 searchCts = new CancellationTokenSource();
                 var oldCts = Interlocked.Exchange(ref _searchCts, searchCts);
-                if (oldCts != null) {
-                    oldCts.Cancel();
-                    oldCts.Dispose();
+                try {
+                    oldCts?.Cancel();
+                    oldCts?.Dispose();
+                } catch (ObjectDisposedException) {
+                }
+
+                CancellationToken token;
+                try {
+                    token = searchCts.Token;
+                } catch (ObjectDisposedException) {
+                    // highly unlikely race, but easy enouh to protect against
+                    return;
                 }
 
                 await _library.VisitNodesAsync(
                     new LibraryNodeVisitor(this, callback, searchValue, _matchMode),
-                    searchCts.Token
+                    token
                 );
             } catch (OperationCanceledException) {
             } catch (Exception ex) when (!ex.IsCriticalException()) {
                 ex.ReportUnhandledException(_serviceProvider, GetType());
             } finally {
                 callback.Done();
+                if (searchCts != null) {
+                    Interlocked.CompareExchange(ref _searchCts, null, searchCts);
```

Comment:
It doesn't need to, because it will only be replaced if it is our current one, and only when we've stopped paying attention to it. But this is the right place to dispose `searchCts`.